flepimop/gempyor_pkg/src/gempyor/NPI/MultiPeriodModifier.py

```diff
@@ -175,6 +175,8 @@ def __get_affected_subpops_grp(self, grp_config):
             affected_subpops_grp = self.subpops
         else:
             affected_subpops_grp = [str(n.get()) for n in grp_config["subpop"]]
+            affected_subpops_grp  = list(set(affected_subpops_grp).intersection(self.subpops))
```

Comment:
```suggestion
            affected_subpops_grp = set(grp_config["subpop"].as_str_seq()).intersection(self.subpops)
```

there's a template getter for string sequences, set accepts an iterable as input.

does this actually need to be an indexable (as opposed to just iterable) container? if not, can just return the set rather than converting to list.

either way, should add the type hint to method signature above.